### DISCUSSION TITLE
Envio de Arquivo PDF

### DISCUSSION THREAD

**Discussion by MrWolfX360** [viniciussanchez/RESTRequest4Delphi#171] [Q&A]
Olá Vinicius, tudo bem?

Estou tentando enviar um arquivo PDF, mas sem sucesso, ele chega no servidor com o conteúdo vazio... Se uso os componentes REST nativos do Delphi, ele vai!!!

Mas gostaria de deixar tudo padronizado, apenas com uma única solução!!!

Sebe me dizer, se pelo RESTRequest4Delphi, eu consigo?

Abraços

**Comment by viniciussanchez** [comment]
Oi, tudo bem e voce?
Consegue sim...
como está fazendo?

**Reply by MrWolfX360** [reply]
Estou bem, também, Graças a Deus!!!

Fiz dessa forma:

procedure TForm1.Button1Click(Sender: TObject);
VAR Arquivo: TMemoryStream;
    sArquivo: WideString;
    R: IResponse;
begin
     IF OpenDialog1.Execute THEN
     begin
          Arquivo:= TMemoryStream.Create;
          Arquivo.LoadFromFile(OpenDialog1.FileName);
          sArquivo:= Base64FromBitmap(Arquivo);
          R:= TRequest.New.BaseURL('http://localhost:8082')
              .Resource('sendpdffile')
              .AddParam('Arquivo',OpenDialog1.FileName)
              .AddParam('Diretorio','PDF')
              .AddParam('FileSend',sArquivo)
              .BasicAuthentication('testparam','testparam')
              .ContentType('application/json')
              .Post;
     end;
end;

tentei assim também, mas não dou certo

procedure TForm1.Button1Click(Sender: TObject);
VAR Arquivo: TMemoryStream;
    sArquivo: WideString;
    R: IResponse;
begin
     IF OpenDialog1.Execute THEN
     begin
          Arquivo:= TMemoryStream.Create;
          Arquivo.LoadFromFile(OpenDialog1.FileName);
          sArquivo:= Base64FromBitmap(Arquivo);
          R:= TRequest.New.BaseURL('http://localhost:8082')
              .Resource('sendpdffile')
              .AddParam('Arquivo',OpenDialog1.FileName)
              .AddParam('Diretorio','PDF')
              .AddParam('FileSend',sArquivo,pkREQUESTBODY)
              .BasicAuthentication('testparam','testparam')
              .ContentType('application/json')
              .Post;
     end;
end;

**Reply by MrWolfX360** [reply]
Com os componentes nativos do Delphi, funciona... 

procedure TForm1.Button2Click(Sender: TObject);
VAR sArquivo: WideString;
    Arquivo: TMemoryStream;
begin
     IF OpenDialog1.Execute THEN
     begin
          //sArquivo:= Memo1.Lines.Text;
          Arquivo:= TMemoryStream.Create;
          Arquivo.LoadFromFile(OpenDialog1.FileName);
          sArquivo:= Base64FromBitmap(Arquivo);
          RequestItensEntrega.Params.Clear;
          RequestItensEntrega.AddParameter('Arquivo',OpenDialog1.FileName);
          RequestItensEntrega.AddParameter('Diretorio','PDF');
          RequestItensEntrega.AddParameter('FileSend',sArquivo);
          RequestItensEntrega.ExecuteAsync(nil, True, True, nil);
     end;
end;

**Reply by viniciussanchez** [reply]
Tenta no arquivo colocar um .AddFile()
E nos demais parâmetros coloque sempre o pkREQUESTBODY, acho que vai funcionar

**Reply by MrWolfX360** [reply]
Não deu certo!!! 
O servidor que recebe esse arquivo, não é meu... E sim de terceiros... 
E ele pedem que o arquivo seja mandado em base64, através do parâmetro "FileSend".

E quanto mando via RR4D, o arquivo fica com 0 K, e segundo o suporte do servidor, o parâmetro "FileSend" vem em branco!!!

**Reply by viniciussanchez** [reply]
Tem alguma documentação essa api para que possamos testar?
